### PR TITLE
🚀 feat(yoga): add config to add typescript environment (web)

### PR DESCRIPTION
**Pull request by matheushdsbr**
[JIRA Issue](https://gympass.atlassian.net/browse/)

## Description 📄

This PR aims to add an environment configuration to start working with typescript in Yoga

## Platforms 📲

- [ ] Web
- [ ] Mobile

## Type of change 🔍

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? 🧪

[Enter the tips to test this PR]

- [ ] Unit Test
- [ ] Snapshot Test

## Checklist: 🔍

- [x] My code follows the contribution guide of this project [Contributing Guide](https://github.com/Gympass/yoga/blob/master/CONTRIBUTING.md)
- [ ] Layout matches design prototype: [FIGMA](https://figma.com/file/YOUR_LINK)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Screenshots 📸

[Upload your screenshots here]

| Before | After |
| ------ | ----- |
|        |       |

### CODE REVIEW

**Review by MicaelRodrigues — APPROVED** [review]

**Comment by flavia-moraes** [comment]
@matheushdsbr take a look at the conflicts that are being highlighted, please?

**Review by MicaelRodrigues — APPROVED** [review]

**Comment by lsantos1873** [comment]
I have a doubt about the generated type files, based on the tests we did, when locally merging this PR with my [PR](https://github.com/gympass/yoga/pull/692). Are the types already being generated as expected (when building the project) or should I have to do changes on my PR?

**Comment by matheushdsbr** [comment]
> I have a doubt about the generated type files, based on the tests we did, when locally merging this PR with my [PR](https://github.com/gympass/yoga/pull/692). Are the types already being generated as expected (when building the project) or should I have to do changes on my PR?

I left a comment on your pull request about the changes you'll need to make

**Review by diogoverde — APPROVED** [review]

File: packages/yoga/package.json [hunk 1]
```diff
@@ -2,13 +2,15 @@
   "name": "@gympass/yoga",
   "version": "7.85.1",
   "description": "Gympass component library",
-  "main": "./src",
+  "main": "src/index.ts",
```
Comment (flavia-moraes) [inline packages/yoga/package.json:5]: Could it impact something on native? Similar to the other error we had with token PR?

Comment (matheushdsbr) [reply packages/yoga/package.json]: good point, I just removed `src/index.ts` and kept `./src`

**Review by lsantos1873 — APPROVED** [review]

**Pull request merged**